src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorVerifierInProcess.cs

```diff
@@ -181,12 +187,33 @@ public EditorVerifierInProcess(TestServices testServices)
 
             if (!RoslynString.IsNullOrEmpty(applyFix))
             {
+                var codeActionLogger = new CodeActionLogger();
+                using var loggerRestorer = WithLogger(AggregateLogger.AddOrReplace(codeActionLogger, Logger.GetLogger(), logger => logger is CodeActionLogger));
```

Comment:
@mavasani `WithWorkspaceChangedHandler` and `WithLogger` are much more difficult to implement in the old test harness. These handlers should provide much better data about operations that lead to code action application failure.